duckduckgo/iOS
# Edit autofill credentials copy buttons and reveal password behaviour changes

Task/Issue URL: https://app.asana.com/0/0/1203776367328731/f
Tech Design URL:
CC:

**Description**:
Makes a variety of changes to the login details page, including a copy button to the username and password fields and changing the password field reveal behaviour (in both view and edit mode). It also slightly simplifies the code, as there is now only one type of CopyableCell. It possibly also fixes some bugs and incorrect padding I encountered, but I haven't double checked that the components I fixed were previously being used in circumstances where these actually cropped up.



**Steps to test this PR**:
1. Go to the details page for a login and test the copy buttons for both username and password work as expected. Make sure it's not possible to press the button in the same touch as selecting the cell (which brings up the menu at the same time), as that's an issue I encountered whilst doing this.
2. Select the password cell and test the show/hide options behave as expected. Pay particular attention to the text in the menu being correct, since that not updating correctly was an issue at one point.
3. In edit mode, test that the password is revealed when making the field active. It should not hide when it is no longer active
4. Test the x buttons on both the username and password fields work as expected
5. Test that changes are saved correctly when editing
6. In view mode, test that the notes or website url fields padding don't do anything weird, since they originally had too much padding as if they had a button 



**Copy Testing**:


**Orientation Testing**:


**Device Testing**:


**OS Testing**:


**Theme Testing**:


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
**When ready for review, remember to post the PR in MM**
